src/Agent.Listener/JobDispatcher.cs

```diff
@@ -466,7 +466,19 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                     // send notification to machine provisioner.
                     var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
                     var accessToken = systemConnection?.Authorization?.Parameters["AccessToken"];
-                    await notification.JobStarted(message.JobId, accessToken, systemConnection.Url);
+                    VariableValue buildId = null;
+                    VariableValue definitionId = null;
+                    if (message.Plan.PlanType == "Build")
+                    {
+                        message.Variables.TryGetValue("build.buildId", out buildId);
+                        message.Variables.TryGetValue("system.definitionId", out definitionId);
+                    }
+                    else if (message.Plan.PlanType == "Release")
+                    {
+                        message.Variables.TryGetValue("release.deploymentId", out buildId);
```

Comment:
release.deploymentId? Why? 
There are releaseId, environmentId, deploymentId, phaseId -> Can you tell me why this is the one that is needed? :) I don't want us to come back because there is a need to add a new ID here in a few weeks time. Let's be sure we've thought this though.

Reply:
We are logging ReleaseDefinitionIdentifier and DeploymentIdentifer under sendjob_taskinstance and I wanted to use the same here. Either ways, we just need globally unique identifier for aggregating on the number of jobs. In OKR queries, for builds we currently use a concat of BuildDefinitionIdentifer + BuildId + JobID as JobID alone is not a globally unique GUID. For releases, we use concat of ReleaseDefinitionIdentifier + DeploymentId + JobId. As long as we have a way to count the total number of unique jobs, we  are good to go 👍